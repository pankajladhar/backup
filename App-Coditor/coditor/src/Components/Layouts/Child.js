import React from "react";

const ChildLayout = ({ content }) => {
  return (
    <div className="home flex h-screen">
      <section className="w-2/6 pt-6 bg-white border-r border-gray-300 flex-shrink-0 border-r border-gray-400">
        {content.leftComponent()}
      </section>
      <section className="border-gray-300 flex-grow">
        {content.rightComponent()}
      </section>
    </div>
  );
};

export default ChildLayout;
