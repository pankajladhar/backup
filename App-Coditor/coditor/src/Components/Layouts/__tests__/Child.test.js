import React from "react";
import renderer from "react-test-renderer";
import Child from "./../Child";

it("should render properly", () => {
  const props = {
    content: {
        leftComponent: () => (<span>Left Component</span>),
        rightComponent: () => (<span>Right Component</span>)
    }
  };
  const tree = renderer.create(<Child {...props}/>);
  expect(tree.toJSON()).toMatchSnapshot();
});
