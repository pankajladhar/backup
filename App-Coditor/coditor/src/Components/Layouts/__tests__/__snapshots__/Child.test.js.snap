// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<div
  className="home flex h-screen"
>
  <section
    className="w-2/6 pt-6 bg-white border-r border-gray-300 flex-shrink-0 border-r border-gray-400"
  >
    <span>
      Left Component
    </span>
  </section>
  <section
    className="border-gray-300 flex-grow"
  >
    <span>
      Right Component
    </span>
  </section>
</div>
`;
